### DISCUSSION TITLE
Add support for creating MR against upstream/forked repository in GitLab when enabling forkProcessing

### DISCUSSION THREAD

**Discussion by valdisrigdon** [renovatebot/renovate#25691] [Suggest an Idea]
### Tell us more.

When running in `forkProcessing: enabled`  mode against GitLab, we would like the Merge Request to be created against the upstream/forked repository, not the discovered repository.  This behavior would match the GitLab UI where it automatically defaults to creating the merge request against the forked repository, not the repository with the branch. 

In our GitLab setup we have a "shared" development forked repository and the upstream repository can only be updated by merge requests.

This would mean that `target_project_id` would be set when creating the Merge Request and that all Merge Request APIs go to the forked repository, not the discovered one.

**Comment by rarkins** [comment]
We don't plan or want to support what you're asking for, because it could lead to chaos. Imagine someone running this way and suddenly generating 10s, 100s or 1000s of unwelcome MRs for the repos they've forked (whose owners have never asked for this).

Instead, you would need:
- #840 to be implemented (it works for GitHub for years, and well)
- Point Renovate to the upstream repo, not the forked one

**Reply by valdisrigdon** [reply]
What if the upstream repository were to have onboarded into renovate? Or a combination of that and checking if the GitLab instance is self-hosted or that the upstream repositories were private?

**Reply by rarkins** [reply]
I prefer to avoid an alternative approach when we already have one which is proven to work well (albeit github-only)
